### DISCUSSION TITLE
Compiling Custom UBL Firmware

### DISCUSSION THREAD

**Discussion by KrazyTurk** [Jyers/Marlin#1802] [Q&A]
To solve a problem I have (with the X axis wanting to move to 245 when the furthest my printer will go without hitting the wheel assembly, I need it to stop at 239 max.
This part I was able to figure out, compile and fix. Great....
Except that there is no UBL Highspeed in the Configuration folder.
I used the bugfix-JyersUI
The video I was following said Ender3v2 branch, but I could not find one. Figured the default branch would work, and it does work for me.
The only thing I would like to do, is to use the High Speed probing so it doesn't retract the probe fully every touch.
I'm not sure where to find the right lines to modify and what to modify them to.
Help Please

**Comment by KrazyTurk** [comment]
Thank you all for answering. I will work this all out when I have a chance.
I got the 4x4 Bilinear working and honestly for that size bed, I think it'll work out just fine.
Using the probe to level my bed tho, did not work very well (the manual level w/ probe)
My right side kept being closer (too close) than my right side, even after mapping the mesh.
Did it with manual movement and paper like the good old days, now she's printing like the darlin' shes always been, but with Jyers options/improvements. Happy for now :)

**Reply by Sonicboom247** [reply]
glad to hear, when you say right side too close, do you mean too close to nozzle or too close to edge?  if it's too close to edge i think you need to add +15 to X_MAX_POS, my (nozzle) now goes WAY off the edge enough that i am considering adding a small wire brush to clean it when the bed goes from back to front.  for now i just hold the brush there and give it a quick scrub.

**Reply by KrazyTurk** [reply]
On the right side I can go to 238. I might even recompile at 237 because at 238 it still touches, it just does not like hit it. I'm also making the point where it probes Go to a 45 margin. I know it's quite a bit away from the springs, but gotta do what we gotta do.
PROBING_MARGIN 45
and 
#define LEVEL_CORNERS_INSET_LFRB { 45, 45, 45, 45 } // (mm) Left, Front, Right, Back insets (Which I think is when you use the manual leveling)
Here is a little video of what happens when I go to 238, you'll see the whole carriage shift and I think that is throwing my BL Touch off.
I had to manually adjust the bed height at the start of a print yesterday, because despite having a very trammed bed and running 4x4 the right side was too close especially the back. I basically went down far enough to where the clicking stopped from the extruder. I'm hoping this fixes this issue.

https://imgur.com/a/ema0ErI

**Reply by KrazyTurk** [reply]
Thank you for you input. My issue was that the +15 was allowing the hotend sled to crash into the right gantry wheel assembly like in the video.
This is all pretty new to me, I don't know what I'm doing, therefore I am here. 
Either way, doing what I did fixed my issue thanks to some of the comments here.
